### ISSUE TITLE
[Question] How does this differ from Passport?

### ISSUE THREAD

**Issue by lonix1** [LionC/express-basic-auth#17]
I was planning to do basic auth using `passport-http` followed by token exchange using `passport-jwt`. But passport hasn't been updated in years and the documentation is a mess.

So I was looking for something else with good documentation, a large user base, typescript support and that is being maintained... and found this! :)

I'm curious how it differs from passport?

**Comment by toverux** [comment]
I don't really know how passport is different on this exactly, but I wouldn't hesitate so much!
At least this library is maintained, has official typings and @LionC announced he was interested in a TypeScript rewrite.
Basic authentication is a very simple thing, you could implement it yourself in a few lines of code. So, choosing one library or the other as not much implications. It won't evolve much either. That's a simple utility that does one job. That explains how this library has only 118 stars but 40k/dl/week, people find it, install it and forget about it -- so you won't be alone, too.
Just use what fits best for you. Ihmo you don't need all the complexity of passport + 1 or 2 `@types` packages for that, that'd be crazy :)

**Comment by lonix1** [comment]
Thanks @toverux! I'm new to this and over the last few days that is exactly what I learned. After I properly understood what BA auth is, I realised that all passport-http is doing is:
- read Authorization header
- give you username/password and ask you if it's okay
- you check db, tell it yes
- it gives you username/password and user object, as `req.user` and "allows" the request

So "basically" (excuse the pun), it does very little! Exactly as you said. But the baggage of using passport is crazy and it's complicated to use...
